### ISSUE TITLE
ImportError: cannot import name 'gcd' from 'fractions' (/usr/lib64/python3.9/fractions.py)

### ISSUE THREAD

**Issue by captcha1** [vispy/vispy#1947]
I upgraded to Fedora 33, which uses python 3.9 and when I run a simple vispy demo, I get an error : 

$ python isocurve.py                                                                                               
Traceback (most recent call last):                                                                                                    
  File "/home/kjh/vispy.d/isocurve.py", line 13, in <module>                                                                          
    import vispy.scene                                                                                                                
  File "/home/kjh/toolroot.d/pythonuserbase.d/lib/python3.9/site-packages/vispy/scene/__init__.py", line 33, in <module>              
    from .visuals import *  # noqa                                                                                                    
  File "/home/kjh/toolroot.d/pythonuserbase.d/lib/python3.9/site-packages/vispy/scene/visuals.py", line 18, in <module>               
    from .. import visuals                                                                                                            
  File "/home/kjh/toolroot.d/pythonuserbase.d/lib/python3.9/site-packages/vispy/visuals/__init__.py", line 14, in <module             >                                                                                                                               
    from .axis import AxisVisual  # noqa
  File "/home/kjh/toolroot.d/pythonuserbase.d/lib/python3.9/site-packages/vispy/visuals/axis.py", line 11, in <module>
    from .visual import CompoundVisual
  File "/home/kjh/toolroot.d/pythonuserbase.d/lib/python3.9/site-packages/vispy/visuals/visual.py", line 94, in <module>
    from .transforms import TransformSystem
  File "/home/kjh/toolroot.d/pythonuserbase.d/lib/python3.9/site-packages/vispy/visuals/transforms/__init__.py", line 10, in <module>
    from .base_transform import BaseTransform  # noqa
  File "/home/kjh/toolroot.d/pythonuserbase.d/lib/python3.9/site-packages/vispy/visuals/transforms/base_transform.py", line 238, in <module>
    from .chain import ChainTransform  # noqa
  File "/home/kjh/toolroot.d/pythonuserbase.d/lib/python3.9/site-packages/vispy/visuals/transforms/chain.py", line 9, in <module>
    from .linear import NullTransform
  File "/home/kjh/toolroot.d/pythonuserbase.d/lib/python3.9/site-packages/vispy/visuals/transforms/linear.py", line 10, in <module>
    from ...geometry import Rect
  File "/home/kjh/toolroot.d/pythonuserbase.d/lib/python3.9/site-packages/vispy/geometry/__init__.py", line 20, in <module>
    from .torusknot import TorusKnot  # noqa
  File "/home/kjh/toolroot.d/pythonuserbase.d/lib/python3.9/site-packages/vispy/geometry/torusknot.py", line 4, in <module>
    from fractions import gcd
ImportError: cannot import name 'gcd' from 'fractions' (/usr/lib64/python3.9/fractions.py)

... a simple change to vispy/geometry/torusknot.py fixes it : 

$ diff vispy/geometry/torusknot.py*
4c4
< #from fractions import gcd
---
> from fractions import gcd
91c91
<         return math.gcd(self._p, self._q)
---
>         return gcd(self._p, self._q)

**Comment by djhoese** [comment]
Looks like this has been deprecated since python 3.5 (https://docs.python.org/3.5/library/fractions.html#fractions.gcd). Your fix seems like the right one. Would you mind making a pull request?

**Comment by captcha1** [comment]
Oops, the diff should look like this : 

... a simple change to vispy/geometry/torusknot.py fixes it : 
```
$ diff vispy/geometry/torusknot.py*
4c4
< #from fractions import gcd
---
> from fractions import gcd
91c91
<         return math.gcd(self._p, self._q)
---
>         return gcd(self._p, self._q)


```

**Comment by HamidKianGaikani** [comment]
Hi,
I have recently started learning bioinformatics and working in Linux environment. 
I have the same problem, but when I read this post and comments, I cannot understand much. I'd highly appreciate it if you could kindly help me. 

the problem is that I downloaded a tool, multiqc, using conda, but when I want to run multiqc, it returns: **cannot import name 'gcd' from 'fractions'
![2021-04-12 11 27 32](https://user-images.githubusercontent.com/56851649/114443221-4758fc80-9b82-11eb-81d4-291740d40db1.jpg)
**

**Comment by djhoese** [comment]
@HamidKianGaikani This is an incompatibility of the `networkx` library with more recent versions of python (actually python 3.5+ apparently). This doesn't have anything to do with vispy or `multiqc`. However, it looks like networkx has also been fixed for quite a while now: https://github.com/networkx/networkx/blob/main/networkx/algorithms/dag.py

Any chance you have an old version of networkx installed? If you can't solve this, please file a bug with the networkx project.

Also it looks like this is fixed for vispy at least in the master branch (not sure if it is released), so I'm closing this.

**Comment by djhoese** [comment]
Fixed by #1914
